controller_manager/src/controller_manager.cpp

```diff
@@ -1131,16 +1131,27 @@ controller_interface::return_type ControllerManager::update()
     rt_controllers_wrapper_.update_and_get_used_by_rt_list();
 
   auto ret = controller_interface::return_type::OK;
-  for (auto loaded_controller : rt_controller_list)
-  {
+  update_loop_counter_ += 1;
+  update_loop_counter_ %= update_rate_;
+
+  for (auto loaded_controller : rt_controller_list) {
     // TODO(v-lopez) we could cache this information
     // https://github.com/ros-controls/ros2_control/issues/153
-    if (is_controller_running(*loaded_controller.c))
-    {
-      auto controller_ret = loaded_controller.c->update();
-      if (controller_ret != controller_interface::return_type::OK)
-      {
-        ret = controller_ret;
+    if (is_controller_running(*loaded_controller.c)) {
+
+      int controller_update_rate = loaded_controller.c->get_update_rate();
```

Comment:
I would add this into CM's structure.

Furthermore, `controller_update_rate` is not clear terminology for this because it is actually "reduction from the base update rate" and not an update rate in "ms" what one could expect from the name.

Reply:
AFAIK rate in ROS has always been defined in `Hz`. This is not reeeeally a reduction from the controller. You can define 100 for CM and 10 for this, you can also define 11

Reply:
Sorry, yeah update rate in `Hz` of course. 

I don't understand the second part. What I think of this is that we have "harmonics" of controllers base frequency, this means, if CM runs at `100 Hz` and we define `10` for controller it will run on `10 Hz`. Isn't it like this?